sources/core/Game.cpp

```diff
@@ -707,15 +707,15 @@ void Game::generateUnsync()
     }
 }
 
-void Game::AddLastMessages(std::stringstream& stream)
+void Game::AddLastMessages(QByteArray& data)
 {
-    stream << std::endl;
+    data.append('\n');
     for (int i = (log_pos_ + 1) % messages_log_.size();
              i != log_pos_;
              i = (i + 1) % messages_log_.size())
     {
-        stream << messages_log_[i].type << " ";
-        stream << messages_log_[i].json.toStdString() << std::endl;
+        data.append(messages_log_[i].type + " ");
+        data.append(messages_log_[i].json + '\n');
```

Comment:
Isn't " " a constant string? Sorry I cannot imagine how " " can be a pointer, so I'm curious. I knew about pointer arithmetic just didn't know " " is a pointer in this case.